minet/cli/twitter/friends.py

```diff
@@ -30,12 +31,40 @@ def twitter_friends_action(namespace, output_file):
         add=REPORT_HEADERS
     )
 
+    loading_bar = tqdm(
+        desc='Retrieving ids',
+        dynamic_ncols=True,
+        total=namespace.total,
+        unit=' line'
+    )
+
     for row, user_id in enricher.cells(namespace.column, with_rows=True):
         all_ids = []
+        next_cursor = 0
+        result = None
 
-        result = wrapper.call('friends.ids', args={'user_id': user_id})
+        wrapper_args = {'user_id': user_id}
+        result = wrapper.call('friends.ids', wrapper_args)
 
         if result is not None:
-            all_ids = result.get('ids', None)
+            all_ids = result.get('ids', [])
+            next_cursor = result.get('next_cursor', None)
+
             for friend_id in all_ids:
                 enricher.writerow(row, [friend_id])
+
+            while next_cursor > 0:
+                wrapper_args['cursor'] = next_cursor
+                result = wrapper.call('friends.ids', wrapper_args)
+
+                if result is not None:
+                    all_ids = result.get('ids', [])
+                    next_cursor = result.get('next_cursor', None)
+
+                    for friend_id in all_ids:
+                        enricher.writerow(row, [friend_id])
+
```

Comment:
Y a un peu deux fois le même code imbriqué, je pense que tu pourrais le réorganiser de sorte à n'avoir la ligne wrapper.call, le test de result is not None, les result.get et le for friend_id une seule fois

Reply:
Et voilà pourquoi python devrait avoir un `do ... while`. Révolution!

Reply:
ça se fait sans souci avec le while, cf https://github.com/medialab/gazouilloire/blob/master/collect_friendships/collect_followers.py#L31-L39
Il faut mettre cursor = -1 dans la première requete pour que twitter le gère

Reply:
Et on veut des labels et des instructions de `jump`. Oui monsieur!

Reply:
T'as qu'à faire de l'assembleur